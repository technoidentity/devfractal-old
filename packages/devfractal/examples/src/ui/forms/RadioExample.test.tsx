import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import { Radio } from './RadioExample'

test('Radio Example', () => {
  const component: ReactTestRenderer = renderer.create(<Radio />)
  expect(component.toJSON()).toMatchSnapshot()
})
