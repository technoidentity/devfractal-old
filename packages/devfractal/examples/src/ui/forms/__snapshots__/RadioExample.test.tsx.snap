// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Example 1`] = `
Array [
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Controlled Radio(
      red
      )
    </h1>
    <div
      className="control"
    >
      <label
        className="radio"
      >
        <input
          checked={true}
          name="radio"
          onChange={[Function]}
          type="radio"
          value="red"
        />
         Red
      </label>
      <label
        className="radio"
      >
        <input
          checked={false}
          name="radio"
          onChange={[Function]}
          type="radio"
          value="green"
        />
         Green
      </label>
      <label
        className="radio"
      >
        <input
          checked={false}
          name="radio"
          onChange={[Function]}
          type="radio"
          value="blue"
        />
         Blue
      </label>
    </div>
  </section>,
  <div
    className="columns"
  >
    <div
      className="column is-narrow"
    >
      <section
        className="section"
      >
        <h1
          className="title is-4"
        >
          Radio button
        </h1>
        <div
          className="field is-horizontal"
        >
          <label
            className="label"
          />
          <div
            className="field-label"
          >
            <label
              className="label"
            >
              Gender
            </label>
          </div>
          <div
            className="field-body"
          >
            <div
              className="control"
            >
              <label
                className="radio"
              >
                <input
                  checked={true}
                  name="gender"
                  onChange={[Function]}
                  type="radio"
                  value="0"
                />
                 Male
              </label>
              <label
                className="radio"
              >
                <input
                  checked={false}
                  name="gender"
                  onChange={[Function]}
                  type="radio"
                  value="1"
                />
                 Female
              </label>
            </div>
          </div>
          <p
            className="help"
          />
        </div>
        <hr />
      </section>
      <section
        className="section"
      >
        <h1
          className="title is-4"
        >
          Radio button with 
          <em>
            female
          </em>
           as default
        </h1>
        <div
          className="field is-horizontal"
        >
          <label
            className="label"
          />
          <div
            className="field-label"
          >
            <label
              className="label"
            >
              Gender
            </label>
          </div>
          <div
            className="field-body"
          >
            <div
              className="control"
            >
              <label
                className="radio"
              >
                <input
                  checked={false}
                  name="gender2"
                  onChange={[Function]}
                  type="radio"
                  value="male"
                />
                 Male
              </label>
              <label
                className="radio"
              >
                <input
                  checked={true}
                  name="gender2"
                  onChange={[Function]}
                  type="radio"
                  value="female"
                />
                 Female
              </label>
            </div>
          </div>
          <p
            className="help"
          />
        </div>
        <hr />
      </section>
      <section
        className="section"
      >
        <h1
          className="title is-4"
        >
          Disabled 
        </h1>
        <div
          className="control"
        >
          <label
            className="radio"
          >
            <input
              checked={true}
              name="going"
              onChange={[Function]}
              type="radio"
              value="0"
            />
             Going
          </label>
          <label
            className="radio"
          >
            <input
              checked={false}
              disabled={true}
              name="going"
              onChange={[Function]}
              type="radio"
              value="1"
            />
             May be
          </label>
        </div>
        <hr />
      </section>
      <section
        className="section"
      >
        <h1
          className="title is-4"
        >
          Controlled Radio
        </h1>
        <div
          className="control"
        >
          <label
            className="radio"
          >
            <input
              checked={false}
              name="color"
              onChange={[Function]}
              type="radio"
              value="red"
            />
             Red
          </label>
          <label
            className="radio"
          >
            <input
              checked={true}
              name="color"
              onChange={[Function]}
              type="radio"
              value="green"
            />
             Green
          </label>
          <label
            className="radio"
          >
            <input
              checked={false}
              name="color"
              onChange={[Function]}
              type="radio"
              value="blue"
            />
             Blue
          </label>
        </div>
        <hr />
      </section>
      <section
        className="section"
      >
        <h1
          className="title is-4"
        >
          Controlled Readonly Radio
        </h1>
        <div
          className="control"
        >
          <label
            className="radio"
          >
            <input
              checked={false}
              name="color2"
              onChange={[Function]}
              readOnly={true}
              type="radio"
              value="red"
            />
             Red
          </label>
          <label
            className="radio"
          >
            <input
              checked={true}
              name="color2"
              onChange={[Function]}
              readOnly={true}
              type="radio"
              value="green"
            />
             Green
          </label>
          <label
            className="radio"
          >
            <input
              checked={false}
              name="color2"
              onChange={[Function]}
              readOnly={true}
              type="radio"
              value="blue"
            />
             Blue
          </label>
        </div>
        <hr />
      </section>
      <section
        className="section"
      >
        <h1
          className="title is-4"
        >
          Uncontrolled Radio
        </h1>
        <div
          className="control"
        >
          <label
            className="radio"
          >
            <input
              checked={true}
              name="color3"
              onChange={[Function]}
              type="radio"
              value="red"
            />
             Red
          </label>
          <label
            className="radio"
          >
            <input
              checked={false}
              name="color3"
              onChange={[Function]}
              type="radio"
              value="green"
            />
             Green
          </label>
          <label
            className="radio"
          >
            <input
              checked={false}
              name="color3"
              onChange={[Function]}
              type="radio"
              value="blue"
            />
             Blue
          </label>
        </div>
      </section>
    </div>
  </div>,
]
`;
